### PR TITLE
Implementation ingredient structure

### PR DESCRIPTION
**Pull request by osvegn**
Pull request link to this issue: #54.

### CODE REVIEW

File: src/Core/Pizza.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                    throw PlazzaEX("There is not enought ingredient in the vector.", Logger::LOW);
+                }
+            }
+        }
+    }
+    lo = tmp;
+    return (lo);
+}
+
+std::vector<Ingredient> &operator-=(std::vector<Ingredient> &lo, std::vector<Ingredient> &ro)
```
Comment (Cavonstavant) [inline src/Core/Pizza.cpp:83]: most of the time when overloading operators you'll want to name parameters "lhs" and "rhs" as in "left hand side" and "right hand side".

**Review by Cavonstavant — APPROVED** [review]
LGTM!

File: src/Core/Pizza.cpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+{
+    std::vector<Ingredient> tmp = lo;
+
+    for (std::size_t i = 0; i < tmp.size(); i++) {
```
[the remark this answers is not in the document]
Comment (osvegn) [reply src/Core/Pizza.cpp:87]: This is a temporary variable, so if an error occurs, only the temporary variable will be change and not the left operent. But if all is good, the left operent will be set to the new value.

File: src/Core/Pizza.cpp (hunk 1 above)
Comment (osvegn) [reply src/Core/Pizza.cpp:83]: I use ro ans lo for left operent and right operent, it works too.

**Review by 31Nathan — APPROVED** [review]
LGTM!

**Pull request merged**